RACK-Ontology/ontology/SOFTWARE.sadl

```diff
@@ -114,17 +117,25 @@ NAMESPACE
     is a COMPONENT_TYPE
     has identifier "NAMESPACE".
 
-COMPONENT
-    (note "An abstract software component")
+SWCOMPONENT
+    (note "A concrete software component")
     is a type of ENTITY.
-    name describes COMPONENT with values of type string.
-    componentType describes COMPONENT with a single value of type COMPONENT_TYPE.
-    valueType (note "The type of this value, if applicable (e.g. for functions or variables).") describes COMPONENT with values of type string.
-    instantiates (note "What logical component (system) does this physical component (code) instantiate or implement?") describes COMPONENT with values of type ENTITY.
-    definedIn (note "Defining entity of this code structure. This is typically a FILE.") describes COMPONENT with values of type ENTITY.
-    mentions (note "A component referenced by this one, e.g., a callee or variable being used.") describes COMPONENT with values of type ENTITY.
-    subcomponentOf (note "A structural sub-component, e.g., a function might be a subcomponent of module.") describes COMPONENT with values of type ENTITY.
-    requirements (note "Low-level requirements that apply to this component.") describes COMPONENT with values of type ENTITY.
-    annotations (note "Analysis annotations that apply to this component.") describes COMPONENT with values of type ENTITY.
-    controlFlowsToUnconditionally (note "Control flows from this component to the target unconditionally.") describes COMPONENT with values of type COMPONENT.
-    controlFlowsToConditionally (note "Control flows from this component to the target conditionally.") describes COMPONENT with values of type COMPONENT.
+
+    componentType (note "Type of component")
+        describes SWCOMPONENT with a single value of type COMPONENT_TYPE.
+
+    valueType (note "The type of this value, if applicable (e.g. for functions or variables).")
+        describes SWCOMPONENT with values of type string.
```

Comment:
range for valueType being a string seems odd. In general it seems like we should avoid strings  for ranges on items outside the basic THING class.  It seems to me it should either be handled like componentType was a special subclass of ENTITY, or the valueType should be moved to a supplemental ontology

Reply:
I envision this being used for capturing a function type signature `int(int, int, char*)` or a C++ variable's type `std::vector<who,knows>` or some other domain specific notion. Do you think these are things that we'd want to capture as an object graph instead?

Reply:
My opinion: we don't want to add entity classes for every possible type signature - that could be an infinite set - nor even all known data types - also perhaps an infinite set. String lets us be specific while providing data that's human-usable.